lib/Config/LocationAwareConfigRepository.php

```diff
@@ -330,6 +341,9 @@ private function searchAndReplaceMissingParameters(array &$data): void
      */
     private function getVarConfigFile(string $key): string
     {
+        if($this->options && array_key_exists('directory', $this->options)) {
+            return $this->options['directory'] . '/' . $key . '.yaml';
+        }
         return $this->storageDirectory . '/' . $key . '.yaml';
```

Comment:
What do you think of: 

```suggestion
        return ($this->options['directory'] ?? $this->storageDirectory) . '/' . $key . '.yaml';
```

or maybe:

```suggestion
        $directory = $this->options['directory'] ?? $this->storageDirectory;

        return $directory . '/' . $key . '.yaml';
```